kernel/src/main/scala/org/typelevel/sbt/TypelevelKernelPlugin.scala

```diff
@@ -48,15 +48,27 @@ object TypelevelKernelPlugin extends AutoPlugin {
       addCommandAlias("tlReleaseLocal", mkCommand(List("reload", "project /", "+publishLocal")))
 
   override def projectSettings = Seq(
-    skip := {
-      skip.value || {
+    (Test / test) := {
+      if ((Test / test / skip).value)
+        ()
+      else (Test / test).value
+    },
+    skipIfIrrelevant(compile),
+    skipIfIrrelevant(test),
+    skipIfIrrelevant(publishLocal),
+    skipIfIrrelevant(publish)
+  )
+
+  private[sbt] def mkCommand(commands: List[String]): String = commands.mkString("; ", "; ", "")
+
+  private[sbt] def skipIfIrrelevant[T](task: TaskKey[T]) = {
```

Comment:
I kept this private, but IMO it's worth exposing. For example http4s could use this for the `unusedCompileDependenciesTest` that was causing problems for the non-Scala 3 modules.

Reply:
Although, many tasks are not natively `skip`pable (like `test`).

Reply:
Yeah, seems worth exposing.